script/check_same_primer_view_components_version_everywhere

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -e
+
+echo "Checking for same primer view components version in Gemfile.lock and frontend/package.json"
+
+versions=$(grep --color=never primer.view.components frontend/package.json Gemfile.lock \
+          | grep --extended-regexp --only-matching '\d+\.\d+\.\d+' \
```

Comment:
@cbliard this would obviously break if we were to do a minor release without the patch level component, e.g. `0.53`. @HDinger do we always include the patch level (`0`) when we do major/minor releases?

Reply:
It would also not work with pre-release versions such as 1.2.3-alpha , however I'm not sure if we ever to pre releases.

Reply:
We are currently publishing canary versions - see workflow here: https://github.com/opf/primer_view_components/actions/runs/12822373410/job/35755182385

```
🦋  info Publishing "@openproject/primer-view-components" at "0.0.0-20250117035659"
🦋  success packages published successfully:
🦋  @openproject/primer-view-components@0.0.0-20250117035659
```

Whether or not we're actively using the canary versions while developing core features, I'm not sure (I've been working mostly linearly so far, so haven't had a need to do this yet). 

@HDinger @bsatarnejad do you know? Do you always release `primer-view-components` before updating call sites in core (as I did with #17569).

Reply:
Yes, [it follows semantic versioning](https://github.com/opf/primer_view_components/blob/main/script/version).
And it won't break, it would just go unnoticed and print something like this:

```
Ok. Primer view components version is  everywhere
```

version is not detected

I could amend it slightly to make the message different if version can't be detected, but frankly that's overkill.

Reply:
> @cbliard this would obviously break if we were to do a minor release without the patch level component, e.g. `0.53`. @HDinger do we always include the patch level (`0`) when we do major/minor releases?

Yes, we do

Reply:
> @HDinger @bsatarnejad do you know? Do you always release `primer-view-components` before updating call sites in core (as I did with #17569).

usually we do it like that, yes. I can imagine, that we want to try things with the canary versions locally but usually not in production.

Reply:
Hi  @myabc I always release Primer-view-components first, then update core with new version both in front-end and backend. Afterward, I make the necessary changes in the relevant places.

Reply:
I updated the script to include the pre-release version if any (the "pre.3-rt12" part in 1.2.3-pre.3-rt12 version), and also to have a better message if version cannot be detected.